.github/workflows/publish-to-test-pypi.yml

```diff
@@ -0,0 +1,56 @@
+# link: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+name: publish distributions 📦 to PyPI and TestPyPI
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-n-publish:
+    if: github.repository_owner == 'insarlab' || github.repository_owner == 'yunjunz'
+
+    name: Build and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
+
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+
+    - name: Publish developed version 📦 to Test PyPI
+      if: github.ref == 'refs/heads/main'
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+        skip_existing: true
+        verbose: true
+
+    - name: Publish released version 📦 to PyPI
+      if: startsWith(github.ref, 'refs/tags/v') && github.repository_owner == 'insarlab'
```

Comment:
Right now, since you just have:
```yaml
on:
  push:
    branches:
      - main
```

I don't thing this will run as as the [`github.ref` context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) will always be the branch one like `refs/heads/main`

If you change the trigger to:
```yaml
on:
  push:
    branches:
      - main
    tags:
      - v*
```

you'll also get `refs/tags/v...`

Reply:
Thanks! I added your suggested change.